NadekoBot.Core/Modules/Searches/SearchesSlashCommands.cs

```diff
@@ -0,0 +1,119 @@
+using AngleSharp;
+using AngleSharp.Dom;
+using AngleSharp.Html.Dom;
+using AngleSharp.Html.Parser;
+using Discord;
+using Discord.Interactions;
+using Discord.WebSocket;
+using Microsoft.Extensions.Caching.Memory;
+using NadekoBot.Common;
+using NadekoBot.Common.Attributes;
+using NadekoBot.Common.Replacements;
+using NadekoBot.Core.Modules.Searches.Common;
+using NadekoBot.Core.Services;
+using System.Collections.Generic;
+using NadekoBot.Extensions;
+using NadekoBot.Modules.Searches.Common;
+using NadekoBot.Modules.Searches.Services;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using SixLabors.ImageSharp;
+using SixLabors.ImageSharp.Drawing.Processing;
+using SixLabors.ImageSharp.PixelFormats;
+using SixLabors.ImageSharp.Processing;
+using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using NadekoBot.Modules.Administration.Services;
+using Serilog;
+using Configuration = AngleSharp.Configuration;
+
+namespace NadekoBot.Modules.Searches
+{
+    public partial class SearchesSlash : NadekoSlashModule<SearchesService>
+    {
+        private readonly IHttpClientFactory _httpFactory;
+
+        private readonly GuildTimezoneService _tzSvc;
+        public SearchesSlash(IHttpClientFactory factory,GuildTimezoneService tzSvc)
+        {
+            _httpFactory = factory;
+            _tzSvc = tzSvc;
+        }
+
+        [NadekoSlash]
+        public async Task Weather([Summary("city","name of the city")]string city,
+        [Summary("country","optional name of country")]string country = "")
+        {
+            if (!await ValidateQuery(ctx.Channel, city).ConfigureAwait(false))
```

Comment:
Are you familiar enough with C# to understand this article? https://devblogs.microsoft.com/dotnet/configureawait-faq/

I don't entirely understand it myself but it seems like we would be fine to not use `.ConfigureAwait(false)` in any of this code. Unfortunately it's already littered in the rest of nadekobot code, but I really don't like this voodoo and we could stop perpetuating it. Leave it out unless the performance of the bot grinds to a halt without it.

Reply:
I didn't understand it either and I cant realistically test performance of the bot in my single member server either so I will leave it there for now. I leave it up to you test the code without it throught the code.

Reply:
Actually its required everywhere as far as i can tell, so you cant remove it at all

Reply:
What makes it required? It deadlocks without it?

Reply:
It hangs without it